src/containers/PipelineRuns/PipelineRuns.js

```diff
@@ -162,21 +216,55 @@ export /* istanbul ignore next */ class PipelineRuns extends Component {
             lowContrast
           />
         )}
-        <Button
-          className="create-pipelinerun-button"
-          iconDescription="Create PipelineRun"
-          renderIcon={Add}
-          onClick={() => this.toggleModal(true)}
-        >
-          Create PipelineRun
-        </Button>
-        <CreatePipelineRun
-          open={this.state.showCreatePipelineRunModal}
-          onClose={() => this.toggleModal(false)}
-          onSuccess={this.handleCreatePipelineRunSuccess}
-          pipelineRef={pipelineName}
-          namespace={selectedNamespace}
-        />
+        <div className="search-bar">
+          <Search
+            placeHolderText="Input a filter e.g. gitRepo:https://github.com/example/helloworld"
+            className="search"
+            onChange={this.handleChange}
+            value={this.state.currentFilterValue}
+            data-testid="filter-search-bar"
+          />
+          <Button
+            className="add-filter-button"
+            iconDescription="Add filter"
+            renderIcon={Add}
+            kind="ghost"
+            onClick={() =>
+              this.handleAddFilter(this.state.currentFilterValue, filters)
+            }
+          >
+            Add filter
+          </Button>
+          <Button
+            className="create-pipelinerun-button"
+            iconDescription="Create PipelineRun"
+            renderIcon={Add}
+            onClick={() => this.toggleModal(true)}
+          >
+            Create PipelineRun
+          </Button>
+        </div>
+        <div className="filters">
+          <div className="filters-title">Filters:</div>
+          {this.props.filters.map(filter => (
+            <Tag
```

Comment:
each component in a list generated from an iterable (`map`, `forEach`, etc.) must have a `key` prop that is unique among its siblings, ideally based on the data (i.e. some property of `filter`)

More detail, including effects of omitting key or using bad values: https://reactjs.org/docs/lists-and-keys.html